src/wallet/feebumper.cpp

```diff
@@ -293,21 +294,35 @@ Result CreateRateBumpTransaction(CWallet& wallet, const uint256& txid, const CCo
     // We cannot source new unconfirmed inputs(bip125 rule 2)
     new_coin_control.m_min_depth = 1;
 
-    constexpr int RANDOM_CHANGE_POSITION = -1;
-    auto res = CreateTransaction(wallet, recipients, RANDOM_CHANGE_POSITION, new_coin_control, false);
-    if (!res) {
-        errors.push_back(Untranslated("Unable to create transaction.") + Untranslated(" ") + util::ErrorString(res));
-        return Result::WALLET_ERROR;
-    }
+    if (recipients.size() > 0) {
+        constexpr int RANDOM_CHANGE_POSITION = -1;
+        auto res = CreateTransaction(wallet, recipients, RANDOM_CHANGE_POSITION, new_coin_control, false);
+        if (!res) {
+            errors.push_back(Untranslated("Unable to create transaction.") + Untranslated(" ") + util::ErrorString(res));
+            return Result::WALLET_ERROR;
+        }
 
-    const auto& txr = *res;
-    // Write back new fee if successful
-    new_fee = txr.fee;
+        const auto& txr = *res;
+        // Write back new fee if successful
+        new_fee = txr.fee;
 
-    // Write back transaction
-    mtx = CMutableTransaction(*txr.tx);
+        // Write back transaction
+        mtx = CMutableTransaction(*txr.tx);
 
-    return Result::OK;
+        return Result::OK;
+    } else {
+        // The transaction only has one output, and it is either detected as change,
+        // or selected as the one to take the fee from.
+        // In that case, just reduce its value to meet the new feerate.
+        new_fee = new_coin_control.m_feerate->GetFee(maxTxSize);
+        temp_mtx.vout[0].nValue += old_fee - new_fee;
+        if (IsDust(temp_mtx.vout[0], wallet.chain().relayDustFee())) {
+            errors.push_back(Untranslated("Single output amount too small"));
+            return Result::MISC_ERROR;
+        }
+        mtx = temp_mtx;
+        return Result::OK;
```

Comment:
This is not contemplating the user changing the outputs for a single change output. E.g. bump tx to send coins back to yourself.

The generated `temp_mtx` uses the same outputs as the previous transaction, so the provided output is discarded. And only deduces the new fee from the previous tx outputs.

Previously, in master, we would just fail on this situation. While here, we return a tx that is not what the user wants.
So, created #27195 to solve the issue in master first.

Reply:
I've added a fix, and adopted your test. The fix is simple enough to do here that I think a separate PR is unnecessary.